risc0/zkvm/src/host/client/prove/opts.rs

```diff
@@ -75,6 +75,15 @@ pub enum ReceiptKind {
     /// Groth16 receipts are proven using Groth16, are constant in size, and are the smallest
     /// available receipt format. A Groth16 receipt can be serialized to a few hundred bytes.
     Groth16,
+
+    #[cfg(feature = "blake3")]
+    /// Request that a Blake3 [Groth16Receipt][crate::Groth16Receipt] be generated.
+    ///
+    /// Groth16 receipts are proven using Groth16, are constant in size, and are the smallest
+    /// available receipt format. A Groth16 receipt can be serialized to a few hundred bytes.
+    /// This variant uses the BLAKE3 hash function and is only compatible with guests that
+    /// have journal size of 32 bytes.
+    Blake3Groth16,
```

Comment:
this too because it is serialized / deserialized I think it better to always compile it in and just error if encountered.